apps/storefront/src/shared/routeList.ts

```diff
@@ -310,6 +313,10 @@ export const getAllowedRoutesWithoutComponent = (globalState: GlobalState): Buye
         });
       }
 
+      if (path === '/purchased-products' && hasPermission) {
+        return true;
+      }
+
```

Comment:
[question]

Why do we need this early return for `path === '/purchased-products'`, isn't it already covered by the `return hasPermission` on the next line?

[suggestion]

```suggestion
```